File: health_track_ui/lib/models/patient_insert_request.dart
class PatientInsertRequest {
  final String firstName;
  final String lastName;
  final DateTime dateOfBirth;
  final String? address;
  final String gender;
  final String? emergencyContact;
  final String? bloodType;
  final String email;
  final String phoneNumber;
  final String userName;
  final bool isActive;

  PatientInsertRequest({
    required this.firstName,
    required this.lastName,
    required this.dateOfBirth,
    this.address,
    required this.gender,
    this.emergencyContact,
    this.bloodType,
    required this.email,
    required this.phoneNumber,
    required this.userName,
    this.isActive = true,
  });

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'dateOfBirth': dateOfBirth.toIso8601String(),
      'address': address,
      'gender': gender,
      'emergencyContact': emergencyContact,
      'bloodType': bloodType,
      'email': email,
      'phoneNumber': phoneNumber,
      'userName': userName,
      'isActive': isActive,
    };
  }
}

File: Application/DTOs/PatientInsertRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.DTOs
{
    public class PatientInsertRequest
    {
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public DateTime DateOfBirth { get; set; }
        public string? Address { get; set; }
        public string Gender { get; set; } = string.Empty;
        public string? EmergencyContact { get; set; }
        public string? BloodType { get; set; }
        public string Email { get; set; } = string.Empty;
        public string PhoneNumber { get; set; } = string.Empty;
        public string UserName { get; set; } = string.Empty;
        public bool IsActive { get; set; } = true;
    }
}
